internal/http/services/appprovider/errors.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package appprovider
```

Comment:
Why are we duplicating the code? Can we refactor the error handling in the ocmd service to a separate package so that all services can use it?

Reply:
OCMD shouldn't expose it's error codes in the first place. They are highly specific for this one particular service (eg. `APIErrorUntrustedService APIErrorCode = "UNTRUSTED_SERVICE"` which maps to `APIErrorUntrustedService: http.StatusForbidden,` ).

I don't see any possibility of adding eg. `appErrorAlreadyExists    appErrorCode = "RESOURCE_ALREADY_EXISTS"` like status code to the OCMD service only to use it in the app provider since this is clearly a violation of concerns.

I agree that we could introduce a common error code mapping but this is out of scope for this PR.

Reply:
@ishank011 how can we continue on this?

Reply:
Sure, let's address it separately. Seems fine for now